tests/unittests/TraceEventsTest.cpp

```diff
@@ -701,6 +701,31 @@ TEST(TraceEventsTest, nestedScopedEventsTerm) {
   ASSERT_GT(durations["five"], durations["four"]);
 }
 
+TEST(TraceEventsTest, TraceLevels) {
+  std::array<TraceLevel, 4> levels = {TraceLevel::NONE, TraceLevel::REQUEST,
+                                      TraceLevel::RUNTIME,
+                                      TraceLevel::OPERATOR};
+  for (auto L : levels) {
+    TraceContext context(L);
+    for (auto evl : levels) {
+      context.logTraceEvent("event", evl);
+    }
+
+    if (L == TraceLevel::NONE) {
+      ASSERT_EQ(context.getTraceEvents().size(), 0);
```

Comment:
in this case expect_eq should be fine, this way we'll also get a chance to test more things down below (before aborting).